src/components/filter/Sue/StatusFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Controller } from 'react-hook-form';
+import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';
+
+import { Label } from '../../Label';
+import { SueStatus } from '../../SUE';
+import { WrapperRow } from '../../Wrapper';
+
+type TData = {
+  status: string;
+  matchingStatuses: string[];
+  iconName: string;
+};
+
+type Props = {
+  containerStyle?: StyleProp<ViewStyle>;
+  control: any;
+  data: TData[];
+  name: string;
+  label: string;
+};
+
+export const StatusFilter = ({ containerStyle, control, data, name, label }: Props) => {
+  return (
+    <View style={(styles.container, containerStyle)}>
+      <Controller
+        name={name}
+        render={({ field: { onChange, value } }) => (
+          <>
+            <Label>{label}</Label>
+            <WrapperRow spaceBetween>
+              {data?.map(
+                (
+                  item: { status: string; matchingStatuses: string[]; iconName: string },
+                  index: number
+                ) => (
+                  <TouchableOpacity onPress={() => onChange(item.status)} key={index}>
```

Comment:
Yes, that's absolutely right, but I didn't think it was important because there is a filter reset button at the bottom. I'm making a note of it and will complete it in the future.

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/1017/commits/9ef2d4e7bdb920db92458b2cd2b76eee56287249